src/vision/status_quo/alan_runs_into_stack_trouble.md

```diff
@@ -0,0 +1,60 @@
+# 😱 Status quo stories: Alan runs into stack allocation trouble
+
+[Alan]: ../characters/alan.md
+[Grace]: ../characters/grace.md
+[Niklaus]: ../characters/niklaus.md
+[Barbara]: ../characters/barbara.md
+
+[Alan runs into stack allocation trouble and is able to fix problems]: TODO
+
+### The problem
+
+Alan is working on a networking service written using async Rust and the tokio async runtime. One day, after many changes to his code base, Alan runs his application and hits an error:
+
+```
+$ .\target\debug\application.exe
+thread 'main' has overflowed its stack
+```
+
+Perplexed, Alan sees if anything with his application works by seeing if he can get output when the `--help` flag is passed, but he has no luck:
+
+```
+$ .\target\debug\application.exe --help
+thread 'main' has overflowed its stack
+```
+
+### Searching for the solution
+
+Having really only ever seen stack overflow issues caused by recursive functions, Alan desperately tries to find the source of the bug but searching through the codebase for recursive functions only to find none. Having learned that Rust favors stack allocation over heap allocation (a concept Alan didn't really need to worry about before), he decided to also search for structs that looked large but none appeared in his code.
```

Comment:
One question here: How would Alan even do this search? I guess you mean he kind of manually inspected, right? (It'd be nice if we had better tooling for this, is part of what I'm thinking...not so much the recursive functions as the large structs)

Reply:
Oh, I guess this came up later! Maybe clarify here anyhow.

Reply:
The first inspection was manually. I can make this clearer.

Reply:
```suggestion
Having really only ever seen stack overflow issues caused by recursive functions, Alan desperately tries to find the source of the bug but searching through the codebase for recursive functions only to find none. Having learned that Rust favors stack allocation over heap allocation (a concept Alan didn't really need to worry about before), he started manually looking through his code, searching for structs that looked "too large"; he wasn't able to find any candidates.
```

Reply:
I added something, just because I'd like to merge this.